shell/platform/darwin/ios/framework/Source/FlutterDartProject.mm

```diff
@@ -143,6 +143,10 @@
       make_mapping_callback(kPlatformStrongDill, kPlatformStrongDillSize);
 #endif  // FLUTTER_RUNTIME_MODE == FLUTTER_RUNTIME_MODE_DEBUG
 
+  if (settings.old_gen_heap_size <= 0) {
```

Comment:
Should probably leave notes here that half is based how much iOS lets a process use before killing it and one million is because the VM flag is in terms of MB.

(Also, the VM thinks MB means 2^20 not 10^6.)

Reply:
Updated